src/main/java/ch/njol/skript/patterns/SkriptPattern.java

```diff
@@ -55,4 +68,24 @@ public String toString() {
 		return first.toFullString();
 	}
 
+	public static String @Nullable[] getKeywords(PatternElement first) {
+		List<String> keywords = new ArrayList<>();
+		PatternElement next = first;
+		while (next != null) {
+			if (next instanceof LiteralPatternElement) {
+				String literal = next.toString().trim();
+				while (literal.contains("  "))
+					literal = literal.replace("  ", " ");
+				keywords.add(literal);
+			} else if (next instanceof GroupPatternElement) {
+				next = ((GroupPatternElement) next).getPatternElement();
+				continue;
+			}
+			next = next.next;
+		}
+		if (keywords.size() == 0)
+			return null;
+		return keywords.toArray(new String[0]);
```

Comment:
Can't you just use a list and check if the list is empty? I mean this seems very cumbersome to me, and nullable arrays are not quite productive. Is there a reason for this approach? 

Other than that, great changes and I will probably use this keyword system in skript-parser as well!